### DISCUSSION TITLE
consensus/Basic-Paxos

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#235] [General]
# consensus/Basic-Paxos

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/consensus/Basic-Paxos.html

**Comment by madeinchinalmc** [comment]
1. Basic Paxos 算法描述 章节，1. Basic Paxos 算法描述段落，接受者需要返回自己接受的值和已经接受的提交号.... 这里的接受者意思是否是决策节点（Acceptor）含义，假如是的话，是否可以依然引用上文专用名词，阅读起来比较别扭。

**Comment by madeinchinalmc** [comment]
下一阶段就是接受阶段 accept(value,n) n是Proposal Number的含义吗（累加的提案号）？ 那么这里的这个value是什么值呢？

**Comment by Solinlee** [comment]
通常是单调递增的数字，相当于乐观锁中的 version，更高的编号意味着更高的优先级
--------
更高编号只能说明相对新旧的记录吧，如果是锁的概念，编号就不对就无法执行的啊。所以这里和优先级没有关系吧

**Reply by isno** [reply]
这段描述不是很严谨，如果有更高的 version，旧的记录无法执行，必须得重新获取最新的 value，更新提案编号，然后继续尝试提交。

这种操作方式，我和前面介绍过的“乐观锁”，有点相似。

**Comment by robben009** [comment]
更正“我们延续“乐观锁”的思路，思考图 6-7 所示并发冲突如何解决”，应该是 “我们延续“乐观锁”的思路，思考图 6-5 所示并发冲突如何解决”

**Comment by robben009** [comment]
更新文案“解决这个问题并复杂，把重试时间随机化，就能减少这种巧合发生”，应改为“解决这个问题并不复杂，把重试时间随机化，就能减少这种巧合发生”  少个不

**Comment by 615lyw** [comment]
"提议者从多数决策者获得了“承诺”（Promise），则“准备阶段”达成。接着，决策者选择提案值：如果决策者的响应中返回了提案值，从中选择编号最高的提案值；如果没有提案值返回，则使用决策者初始提案值。" 这里应该是"接着，提议者 (Proposer) 选择提案值"吧
